### PR TITLE
AM-1: Show Recipes flow

### PR DESCRIPTION
**Pull request by VoisaAndreea**

### CODE REVIEW

**Comment by iulianpopescu** [comment]
Please add .idea folder to .gitignore

File: FoodSuggestions/app/src/main/java/com/example/foodsuggestions/adapters/RecipeAdapter.java [hunk 1]
```diff
@@ -44,39 +49,55 @@ public ViewHolder(View itemView) {
             nameRecipe = (TextView) itemView.findViewById(R.id.idTitleRe);
             imageRecipe = (ImageView) itemView.findViewById(R.id.idImageRe);
 
-            itemView.setOnClickListener(this);
```
Comment (iulianpopescu) [inline FoodSuggestions/app/src/main/java/com/example/foodsuggestions/adapters/RecipeAdapter.java:47]: Please set the click listener in onCreateViewHolder. See https://developer.android.com/reference/androidx/recyclerview/widget/RecyclerView.ViewHolder#getBindingAdapterPosition()

File: FoodSuggestions/app/src/main/java/com/example/foodsuggestions/fragments/IngredientsFragment.java [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+    @Override
+    public View onCreateView(LayoutInflater inflater, ViewGroup container,
+                             Bundle savedInstanceState) {
+        // Inflate the layout for this fragment
+        View view = inflater.inflate(R.layout.fragment_ingredients, container, false);
+        RecyclerView recyclerView = view.findViewById(R.id.idListIngredients);
+
+
+        Bundle data = getArguments();
+        ArrayList<Ingredients> gson =
```
Comment (iulianpopescu) [inline FoodSuggestions/app/src/main/java/com/example/foodsuggestions/fragments/IngredientsFragment.java]: please rename variable.

File: FoodSuggestions/app/src/main/java/com/example/foodsuggestions/fragments/IngredientsFragment.java [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+                new Gson().fromJson(data.getString("ing"),
+                new TypeToken<ArrayList<Ingredients>>(){}.getType());
+
+
+        IngredientsAdapter ingredientsAdapter = new IngredientsAdapter(view.getContext(), gson);
+        RecyclerView.LayoutManager manager = new LinearLayoutManager(view.getContext());
+            recyclerView.setLayoutManager(manager);
```
Comment (iulianpopescu) [inline FoodSuggestions/app/src/main/java/com/example/foodsuggestions/fragments/IngredientsFragment.java:36]: Please indent code properly.

File: FoodSuggestions/app/src/main/java/com/example/foodsuggestions/fragments/InstructionFragment.java [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+//                    new ArrayAdapter<>(view.getContext(), R.layout.row_instruction,
+//                            Collections.singletonList(data.getString("summary")));
+//            listView.setAdapter(listAdapter);
+
+//            InstructionAdapter instructionAdapter = new InstructionAdapter(view.getContext(), data.getString("summary"));
+//            RecyclerView.LayoutManager manager = new LinearLayoutManager(view.getContext());
+//            recyclerView.setLayoutManager(manager);
+//            recyclerView.setHasFixedSize(true);
+//            recyclerView.setAdapter(instructionAdapter);
+            }
```
Comment (iulianpopescu) [inline FoodSuggestions/app/src/main/java/com/example/foodsuggestions/fragments/InstructionFragment.java:46]: please indent code.

File: FoodSuggestions/app/src/main/java/com/example/foodsuggestions/main/ContActivity.java [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+
+        BottomNavigationView bottomNavigationView = findViewById(R.id.idNavbar);
+        bottomNavigationView.setSelectedItemId(R.id.nav_account);
+        bottomNavigationView.setOnNavigationItemSelectedListener(new BottomNavigationView.OnNavigationItemSelectedListener() {
+            @Override
+            public boolean onNavigationItemSelected(@NonNull MenuItem item) {
+                switch (item.getItemId()){
+                    case R.id.nav_home:
+                        startActivity(new Intent(ContActivity.this, HomeActivity.class));
+                        overridePendingTransition(0,0);
```
Comment (iulianpopescu) [inline FoodSuggestions/app/src/main/java/com/example/foodsuggestions/main/ContActivity.java:28]: Please investigate to see if it's needed.

File: FoodSuggestions/app/src/main/java/com/example/foodsuggestions/main/RecipesActivity.java [hunk 6]
```diff
@@ -0,0 +1,155 @@
...
+import retrofit2.Response;
+
+public class RecipesActivity extends AppCompatActivity {
+
+    private List<Recipe> listRecipe;
+    private RecyclerView recyclerView;
+    private RecipeAdapter recipeAdapter;
+
+    private ArrayList<Object> objectArrayList;
+    BottomNavigationView bottomNavigationView;
```
Comment (iulianpopescu) [inline FoodSuggestions/app/src/main/java/com/example/foodsuggestions/main/RecipesActivity.java]: could be private.

File: FoodSuggestions/app/src/main/java/com/example/foodsuggestions/main/ShowRecipe.java [hunk 7]
```diff
@@ -0,0 +1,241 @@
...
+import com.example.foodsuggestions.fragments.FeedbackFragment;
+import com.example.foodsuggestions.fragments.IngredientsFragment;
+import com.example.foodsuggestions.fragments.InstructionFragment;
+import com.example.foodsuggestions.fragments.RecipeFragment;
+import com.google.android.material.tabs.TabLayout;
+
+import org.json.JSONException;
+import org.json.JSONObject;
+
+public class ShowRecipe extends AppCompatActivity {
```
Comment (iulianpopescu) [inline FoodSuggestions/app/src/main/java/com/example/foodsuggestions/main/ShowRecipe.java]: Please add suffix "Activity"

File: FoodSuggestions/app/src/main/java/com/example/foodsuggestions/retrofit/ServiceProvider.java [hunk 8]
```diff
@@ -0,0 +1,31 @@
+package com.example.foodsuggestions.retrofit;
+
+import retrofit2.Retrofit;
+import retrofit2.converter.gson.GsonConverterFactory;
+
+public class ServiceProvider {
+    private RecipeAPI recipeAPI;
+
+    public ServiceProvider() {
```
Comment (iulianpopescu) [inline FoodSuggestions/app/src/main/java/com/example/foodsuggestions/retrofit/ServiceProvider.java]: Please make it private.

File: FoodSuggestions/app/src/main/res/layout/activity_main.xml [hunk 9]
```diff
@@ -1,21 +1,67 @@
 <?xml version="1.0" encoding="utf-8"?>
-<androidx.constraintlayout.widget.ConstraintLayout xmlns:android="http://schemas.android.com/apk/res/android"
+<LinearLayout xmlns:android="http://schemas.android.com/apk/res/android"
     xmlns:app="http://schemas.android.com/apk/res-auto"
```
Comment (iulianpopescu) [inline FoodSuggestions/app/src/main/res/layout/activity_main.xml:3]: Please remove attributes related to ConstraintLayout.

File: FoodSuggestions/app/src/main/res/layout/activity_recipes.xml [hunk 10]
```diff
@@ -1,15 +1,15 @@
 <?xml version="1.0" encoding="utf-8"?>
-<androidx.constraintlayout.widget.ConstraintLayout xmlns:android="http://schemas.android.com/apk/res/android"
+<RelativeLayout xmlns:android="http://schemas.android.com/apk/res/android"
     xmlns:app="http://schemas.android.com/apk/res-auto"
```
Comment (iulianpopescu) [inline FoodSuggestions/app/src/main/res/layout/activity_recipes.xml:3]: Remove unused attributes from ConstraintLayout.

File: FoodSuggestions/app/src/main/res/layout/activity_register.xml [hunk 11]
```diff
@@ -0,0 +1,144 @@
...
+    <EditText
+        android:id="@+id/idRName"
+        android:layout_width="match_parent"
+        android:layout_height="wrap_content"
+        android:ems="10"
+        android:hint="Full Name"
+        android:inputType="text"
+        android:maxLines="1"
+        android:layout_marginStart="25sp"
+        android:layout_marginEnd="25sp"
```
Comment (iulianpopescu) [inline FoodSuggestions/app/src/main/res/layout/activity_register.xml]: If marginStart == marginEnd (marginTop == marginBottom), we can use marginHorizontal = 5dp || marginVertical = 5dp. The same applies also for paddings.

File: FoodSuggestions/app/src/main/res/layout/activity_search.xml [hunk 12]
```diff
@@ -0,0 +1,36 @@
...
+    android:layout_height="match_parent"
+    tools:context=".main.SearchActivity">
+
+    <TextView
+        android:id="@+id/textView4"
+        android:layout_width="wrap_content"
+        android:layout_height="wrap_content"
+        android:layout_marginStart="145dp"
+        android:layout_marginTop="282dp"
+        android:text="SEARCH ACTIVITY"
```
Comment (iulianpopescu) [inline FoodSuggestions/app/src/main/res/layout/activity_search.xml]: please move to strings.xml

File: FoodSuggestions/app/src/main/res/layout/activity_search.xml [hunk 13]
```diff
@@ -0,0 +1,36 @@
...
+
+    <com.google.android.material.bottomnavigation.BottomNavigationView
+        android:id="@+id/idNavbar"
+        android:layout_width="match_parent"
+        android:layout_height="wrap_content"
+        android:layout_alignParentBottom="true"
+        android:layout_margin="30dp"
+        android:elevation="2dp"
+        tools:ignore="MissingConstraints"
+        tools:layout_editor_absoluteX="67dp"
```
Comment (iulianpopescu) [inline FoodSuggestions/app/src/main/res/layout/activity_search.xml]: please remove Constraintlayout related attributes.

File: FoodSuggestions/app/src/main/res/layout/fragment_ingredients.xml [hunk 14]
```diff
@@ -0,0 +1,13 @@
+<?xml version="1.0" encoding="utf-8"?>
+<FrameLayout xmlns:android="http://schemas.android.com/apk/res/android"
+    xmlns:tools="http://schemas.android.com/tools"
+    android:layout_width="match_parent"
+    android:layout_height="match_parent"
+    tools:context=".fragments.IngredientsFragment">
+
+        <androidx.recyclerview.widget.RecyclerView
```
Comment (iulianpopescu) [inline FoodSuggestions/app/src/main/res/layout/fragment_ingredients.xml]: We can have directly RecyclerView as root.

File: FoodSuggestions/app/src/main/res/layout/fragment_instruction.xml [hunk 15]
```diff
@@ -0,0 +1,20 @@
...
+    xmlns:tools="http://schemas.android.com/tools"
+    android:layout_width="match_parent"
+    android:layout_height="match_parent"
+    tools:context=".fragments.InstructionFragment">
+
+    <androidx.recyclerview.widget.RecyclerView
+        android:id="@+id/idListInstructions"
+        android:layout_width="match_parent"
+        android:layout_height="match_parent"
+        android:layout_marginStart="1dp"
```
Comment (iulianpopescu) [inline FoodSuggestions/app/src/main/res/layout/fragment_instruction.xml]: When all margins are equal, we can use the "android:margin" attribute.

File: FoodSuggestions/app/src/main/java/com/example/foodsuggestions/fragments/IngredientsFragment.java (hunk 2 above)
Comment (VoisaAndreea) [reply FoodSuggestions/app/src/main/java/com/example/foodsuggestions/fragments/IngredientsFragment.java]: Fixed

File: FoodSuggestions/app/src/main/java/com/example/foodsuggestions/fragments/IngredientsFragment.java (hunk 3 above)
Comment (VoisaAndreea) [reply FoodSuggestions/app/src/main/java/com/example/foodsuggestions/fragments/IngredientsFragment.java:36]: Fixed

File: FoodSuggestions/app/src/main/java/com/example/foodsuggestions/main/ContActivity.java (hunk 5 above)
Comment (VoisaAndreea) [reply FoodSuggestions/app/src/main/java/com/example/foodsuggestions/main/ContActivity.java:28]: Done

File: FoodSuggestions/app/src/main/java/com/example/foodsuggestions/main/RecipesActivity.java (hunk 6 above)
Comment (VoisaAndreea) [reply FoodSuggestions/app/src/main/java/com/example/foodsuggestions/main/RecipesActivity.java]: Fixed

File: FoodSuggestions/app/src/main/java/com/example/foodsuggestions/main/ShowRecipe.java (hunk 7 above)
Comment (VoisaAndreea) [reply FoodSuggestions/app/src/main/java/com/example/foodsuggestions/main/ShowRecipe.java]: Fixed

File: FoodSuggestions/app/src/main/java/com/example/foodsuggestions/retrofit/ServiceProvider.java (hunk 8 above)
Comment (VoisaAndreea) [reply FoodSuggestions/app/src/main/java/com/example/foodsuggestions/retrofit/ServiceProvider.java]: Fixed

File: FoodSuggestions/app/src/main/res/layout/activity_search.xml (hunk 12 above)
Comment (VoisaAndreea) [reply FoodSuggestions/app/src/main/res/layout/activity_search.xml]: Fixed

File: FoodSuggestions/app/src/main/res/layout/fragment_ingredients.xml (hunk 14 above)
Comment (VoisaAndreea) [reply FoodSuggestions/app/src/main/res/layout/fragment_ingredients.xml]: Fixed

File: FoodSuggestions/app/src/main/res/layout/fragment_instruction.xml (hunk 15 above)
Comment (VoisaAndreea) [reply FoodSuggestions/app/src/main/res/layout/fragment_instruction.xml]: Fixed

File: FoodSuggestions/app/src/main/res/layout/activity_search.xml (hunk 13 above)
Comment (VoisaAndreea) [reply FoodSuggestions/app/src/main/res/layout/activity_search.xml]: Fixed

File: FoodSuggestions/app/src/main/res/layout/activity_recipes.xml (hunk 10 above)
Comment (VoisaAndreea) [reply FoodSuggestions/app/src/main/res/layout/activity_recipes.xml:3]: Fixed

File: FoodSuggestions/app/src/main/res/layout/activity_main.xml (hunk 9 above)
Comment (VoisaAndreea) [reply FoodSuggestions/app/src/main/res/layout/activity_main.xml:3]: Fixed

File: FoodSuggestions/app/src/main/res/layout/activity_register.xml (hunk 11 above)
Comment (VoisaAndreea) [reply FoodSuggestions/app/src/main/res/layout/activity_register.xml]: Fixed

**Pull request merged**